### DISCUSSION TITLE
Start from nobody

### DISCUSSION THREAD

**Discussion by Venique** [shadowsocks/shadowsocks-rust#991] [Q&A]
Hello!

How I can start ssserver from different user (e.g. nobody)?

For now I'm using CentOS 8.4, shadowsocks-rust got from Snap, which automatically generated snap.shadowsocks-rust.ssserver-daemon.service where I've added "User = nobody". But after starting daemon with edited service I get in the logs:

`shadowsocks-rust.ssserver-daemon[7326]: cmd_run.go:1044: WARNING: cannot create user data directory: cannot create snap home dir: mkdir /snap: permission denied`

It seems everything works fine by the way, but I don't know which errors can follow because of lack of home dir. So, maybe there is some other easy way to start service from non-root user?

**Comment by zonyitoo** [comment]
I think the log you have pasted was from Snap.

**Reply by Venique** [reply]
May be, I got this when ssh "snap.shadowsocks-rust.ssserver-daemon":
```
● snap.shadowsocks-rust.ssserver-daemon.service - Service for snap application shadowsocks-rust.ssserver-daemon
   Loaded: loaded (/etc/systemd/system/snap.shadowsocks-rust.ssserver-daemon.service; enabled; vendor preset: disabled)
   Active: active (running) since Sun 2022-10-30 21:11:05 GMT; 3s ago
 Main PID: 3784 (ssserver)
    Tasks: 4 (limit: 9245)
   Memory: 1.0M
   CGroup: /system.slice/snap.shadowsocks-rust.ssserver-daemon.service
           └─3784 /snap/shadowsocks-rust/275/bin/ssserver

oct 30 21:11:05 xxx systemd[1]: Started Service for snap application shadowsocks-rust.ssserver-daemon.
oct 30 21:11:05 xxx shadowsocks-rust.ssserver-daemon[3784]: cmd_run.go:1044: WARNING: cannot create user data directory: cannot create snap home dir: mkdir /snap: permission denied
oct 30 21:11:05 xxx shadowsocks-rust.ssserver-daemon[3784]: loading default config "/etc/shadowsocks-rust/config.json"
oct 30 21:11:05 xxx shadowsocks-rust.ssserver-daemon[3784]: 2022-10-30T21:11:05.206759360+03:00 INFO  shadowsocks server 1.14.3 build 2022-04-04T17:08:31.348428525+00:00
oct 30 21:11:05 xxx shadowsocks-rust.ssserver-daemon[3784]: 2022-10-30T21:11:05.208194556+03:00 INFO  shadowsocks tcp server listening on XXX.XXX.XXX.XXX:XXX, inbound address XXX.XXX.XXX.XXX:>
```
Anyway, does lack of snap home dir while "User=nobody" in service configuration enabled somehow affect proper execution of shadowsocks?

**Reply by zonyitoo** [reply]
```
oct 30 21:11:05 xxx shadowsocks-rust.ssserver-daemon[3784]: cmd_run.go:1044: WARNING: cannot create user data directory: cannot create snap home dir: mkdir /snap: permission denied
```

This line of log is not printed by shadowsocks-rust. This is a Rust project, we don't have `cmd_run.go`.
